### PR TITLE
Adds a default Make task that builds locally in release mode

### LINKED ISSUE #13: Makefile default should build for my platform
When I ran `make`, it started running a release build for Linux on my Mac ⁉️ 

It's because that task is simply the first in the Makefile.

I'd expect `make` alone to build whatever is appropriate for my system.

### PR DESCRIPTION
**Pull request by colindean**
Fixes #13

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -1,3 +1,15 @@
+.PHONY: all
+local:
+	cargo build --release
+	strip target/release/miniserve
+	upx target/release/miniserve
```
Comment (svenstaro) [inline Makefile]: I suppose we shouldn't have that for local stuff and neither the strip.

Comment (colindean) [reply Makefile]: That's fair. Stripping would be a logical part of a `make install`, though. I don't want to get into setting up the stuff for `$PREFIX` to do that.

File: Makefile [hunk 2]
```diff
@@ -1,3 +1,13 @@
+.PHONY: all
```
Comment (svenstaro) [inline Makefile]: Shouldn't the `.PHONY` be `local` instead of `all`?

Comment (colindean) [reply Makefile]: Yes, but it's also better to have all phony tasks under one `.PHONY`, so I did that, too.

File: Makefile [hunk 3]
```diff
@@ -1,17 +1,25 @@
...
 	strip target/x86_64-pc-windows-gnu/release/miniserve.exe
 	upx target/x86_64-pc-windows-gnu/release/miniserve.exe
 
-.PHONY: build-apple
 build-apple:
 	cargo build --target x86_64-apple-darwin --release
 	strip target/x86_64-apple-darwin/release/miniserve
 	upx target/x86_64-apple-darwin/release/miniserve
+
+.PHONY: local run build-apple build-win build-linux
```
Comment (svenstaro) [inline Makefile]: I try to conform to Clark Grubb's Makefile style guide. See [here](http://clarkgrubb.com/makefile-style-guide#phony-targets).

Comment (colindean) [reply Makefile]: Ah, OK. I'll revert my commit but I'll have to wait until I'm done with some other work to do tonight. I wish GitHub's interface had a revert 😩

**Comment by colindean** [comment]
Fixed.

**Comment by svenstaro** [comment]
Great stuff, thanks!

**Pull request merged**